uint8 no_shift 
uint8 no_perioda 
uint32 no_resi 
uint32 no_kspt 
uint32 no_plt 